File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>

using namespace std;

bool cmp(int a, int b) {    
    string c = to_string(a);
    string d = to_string(b);
    
    return c+d > d+c;
}

string solution(vector<int> numbers) {
    string answer = "";
    
    sort(numbers.begin(), numbers.end(), cmp);
    
    for (int i=0;i<numbers.size();i++) {
        answer += to_string(numbers[i]);
    }
    
    if (answer[0] == '0') {
        while(1) {
            if (answer[0] != '0') break;
            answer.erase(answer.begin());
        }
        answer += "0";
    }
    
    return answer;
}

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.java
import java.util.*;

class Solution {
    public String solution(int[] numbers) {
        String[] strs = new String[numbers.length];
        
        for (int i=0;i<numbers.length;++i) {
            strs[i] = String.valueOf(numbers[i]);
        }
        
        Arrays.sort(strs, (a,b) -> {
            return (b+a).compareTo(a+b);
        });
        
        if (strs[0].equals("0")) return "0";
        
        String answer = "";
        for (int i=0;i<strs.length;++i) {
            answer += strs[i];
        }
        
        return answer;
    }
}